### PR TITLE
Damage on forced uncuff

### PR DESCRIPTION
**Pull request by noudoit**
## About the PR

This PR aims to stop people from spam uncuffing - the damage only applies to the person wearing and spamming the uncuff button it does not hurt to take cuffs off someone else.
it also adds a log for uncuffing.
it is 3 blunt damage per uncuff attempt.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: Attempting to uncuff yourself damages you slightly

### CODE REVIEW

File: Content.Shared/Cuffs/Components/CuffableComponent.cs [hunk 1]
```diff
@@ -38,6 +38,13 @@ public sealed class CuffableComponent : Component
     /// </summary>
     [DataField("canStillInteract"), ViewVariables(VVAccess.ReadWrite)]
     public bool CanStillInteract = true;
+
+    /// <summary>
+    /// Damage is applied to someone when they try to uncuff themselves.
+    /// </summary>
+    [DataField("damageOnResist"), ViewVariables(VVAccess.ReadWrite)]
+    public float DamageOnResist = 3f;
```
Comment (metalgearsloth) [inline Content.Shared/Cuffs/Components/CuffableComponent.cs:46]: DamageSpecifier.

File: Content.Shared/Cuffs/SharedCuffableSystem.cs [hunk 2]
```diff
@@ -50,6 +51,7 @@ public abstract class SharedCuffableSystem : EntitySystem
         [Dependency] private readonly SharedInteractionSystem _interaction = default!;
         [Dependency] private readonly SharedPopupSystem _popup = default!;
         [Dependency] private readonly SharedTransformSystem _transform = default!;
+        [Dependency] private readonly DamageableSystem _damageSystem = default!;
```
Comment (metalgearsloth) [inline Content.Shared/Cuffs/SharedCuffableSystem.cs:54]: Alphabetical please.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by DadeKuma** [comment]
This seems unrealistic, you spam the button 34 times and you crit yourself? From wearing cuffs?

If the problem is the spammed popup message we should make it local to the wearer.

**Comment by Skarletto** [comment]
> This seems unrealistic, you spam the button 34 times and you crit yourself? From wearing cuffs?
> 
> If the problem is the spammed popup message we should make it local to the wearer.

trying to uncuff yourself very fast 34 times in less than 30 seconds is very realistic and will not hurt your hands at all

**Comment by EmoGarbage404** [comment]
this really needs better feedback. I could imagine people taking like 30 damage before they realize anythings wrong, since nothing is communicating to them that they are hurting themselves (also forget the "buh in real life it would hurt" because no one is going to think of that)

I think there's also the implication of prisoners being able to crit themselves on will, which seems like it might be an issue? I could imagine that it might be annoying if someone is critting themselves and forcing you to drag them around the moment you arrest them.

**Comment by Aisu9** [comment]
> this really needs better feedback. I could imagine people taking like 30 damage before they realize anythings wrong, since nothing is communicating to them that they are hurting themselves (also forget the "buh in real life it would hurt" because no one is going to think of that)
> 
> I think there's also the implication of prisoners being able to crit themselves on will, which seems like it might be an issue? I could imagine that it might be annoying if someone is critting themselves and forcing you to drag them around the moment you arrest them.

stamina damage? if they stam crit themselves in X attempt then they'll see something is hurting them

**Comment by 0x6273** [comment]
New sec meta is to beat prisoners until they are 3 damage away from crit

**Comment by DadeKuma** [comment]
I think prisoners need a chance to not show that they are removing cuffs. It will also solve this problem as it doesn't make sense to spam the action.

- 30% chance of globally showing "Player is trying to remove the cuffs..."
- 70% of not show any message, or just to the wearer: "You are trying to secretly remove the handcuffs..."

**Comment by noudoit** [comment]
> this really needs better feedback. I could imagine people taking like 30 damage before they realize anythings wrong, since nothing is communicating to them that they are hurting themselves (also forget the "buh in real life it would hurt" because no one is going to think of that)
> 
> I think there's also the implication of prisoners being able to crit themselves on will, which seems like it might be an issue? I could imagine that it might be annoying if someone is critting themselves and forcing you to drag them around the moment you arrest them.

if they do this just leave them perm dead thats the point.

**Comment by noudoit** [comment]
> > this really needs better feedback. I could imagine people taking like 30 damage before they realize anythings wrong, since nothing is communicating to them that they are hurting themselves (also forget the "buh in real life it would hurt" because no one is going to think of that)
> > I think there's also the implication of prisoners being able to crit themselves on will, which seems like it might be an issue? I could imagine that it might be annoying if someone is critting themselves and forcing you to drag them around the moment you arrest them.
> 
> if they do this just leave them perm dead thats the point.

they shouldnt be trying to escape that actively and if they are they are most likely spamming it.

**Comment by noudoit** [comment]
> this really needs better feedback. I could imagine people taking like 30 damage before they realize anythings wrong, since nothing is communicating to them that they are hurting themselves (also forget the "buh in real life it would hurt" because no one is going to think of that)
> 
> I think there's also the implication of prisoners being able to crit themselves on will, which seems like it might be an issue? I could imagine that it might be annoying if someone is critting themselves and forcing you to drag them around the moment you arrest them.

i agree with making it visable - notifiable that you are hurting yourself though.

**Comment by Emisse** [comment]
funney

**Pull request merged**